app/services/search/postgres/chat_channel_membership.rb

```diff
@@ -0,0 +1,61 @@
+module Search
+  module Postgres
+    class ChatChannelMembership
+      ATTRIBUTES = %w[
+        chat_channel_memberships.id
+        chat_channel_memberships.chat_channel_id
+        chat_channel_memberships.last_opened_at
+        chat_channel_memberships.status
+        chat_channel_memberships.user_id
+        chat_channels.channel_name
+        chat_channels.discoverable
+        chat_channels.last_message_at
+        chat_channels.slug
+        chat_channels.status
+        users.name
+        users.profile_image
+        users.username
+      ].freeze
+      private_constant :ATTRIBUTES
+
+      # TODO: @mstruve: When we want to allow people like admins to search ALL
+      # memberships this will need to change
+      PERMITTED_STATUSES = %w[
+        active
+        joining_request
+      ].freeze
+
+      DEFAULT_PER_PAGE = 30
+      private_constant :DEFAULT_PER_PAGE
+
+      MAX_PER_PAGE = 60 # to avoid querying too many items, we set a maximum amount for a page
+      private_constant :MAX_PER_PAGE
+
+      def self.search_documents(user_ids: [], page: 0, per_page: DEFAULT_PER_PAGE)
```

Comment:
Is an empty list of `user_ids` intended to be a valid input?

Reply:
+1, though unlikely it happens on the frontend, we should probably bail if `user_ids` is empty, no need to issue a query. I guess an empty hash is the proper escape route?

Reply:
Nope - good catch! Removed the default value of `[]`.